components/tabs/README.md

```diff
@@ -37,77 +44,217 @@ Tabs are used to present related information in mutually exclusive panels, allow
+  import '@brightspace-ui/core/components/tabs/tabs.js';
+  import '@brightspace-ui/core/components/tabs/tab-panel.js';
+</script>
+
+<d2l-tabs text="Courses">
+  <d2l-tab id="all" text="All" slot="tabs" selected></d2l-tab>
+  <d2l-tab-panel labelled-by="all" slot="panels">Tab content for All</d2l-tab-panel>
+  <d2l-tab id="biology" text="Biology" slot="tabs"></d2l-tab>
+  <d2l-tab-panel labelled-by="biology" slot="panels">Tab content for Biology</d2l-tab-panel>
+  <d2l-tab id="chemistry" text="Chemistry" slot="tabs"></d2l-tab>
+  <d2l-tab-panel labelled-by="chemistry" slot="panels">Tab content for Chemistry</d2l-tab-panel>
+  <d2l-tab id="earth-sciences" text="Earth Sciences" slot="tabs"></d2l-tab>
+  <d2l-tab-panel labelled-by="earth-sciences" slot="panels">Tab content for Earth Sciences</d2l-tab-panel>
+  <d2l-tab id="physics" text="Physics" slot="tabs"></d2l-tab>
+  <d2l-tab-panel labelled-by="physics" slot="panels">Tab content for Physics</d2l-tab-panel>
+  <d2l-tab id="math" text="Math" slot="tabs"></d2l-tab>
+  <d2l-tab-panel labelled-by="math" slot="panels">Tab content for Math</d2l-tab-panel>
+  <d2l-tab id="community" text="Community" slot="tabs"></d2l-tab>
+  <d2l-tab-panel labelled-by="community" slot="panels">Tab content for Community</d2l-tab-panel>
+</d2l-tabs>
+```
+
+<!-- docs: start hidden content -->
+### Properties
+| Property | Type | Description |
+|--|--|--|
+| `id` | String, required | Unique identifier for the tab |
+| `text` | String, required | The text used for the tab and for labelling the corresponding panel |
+| `selected` | Boolean | Use to select the tab |
+
+### Events
+- `d2l-tab-content-change`: Dispatched when the text attribute is changed. Triggers virtual scrolling calculations in parent `d2l-tabs`.
+- `d2l-tab-selected`: Dispatched when a tab is selected
+<!-- docs: end hidden content -->
+
+### Removing a Tab
+
+A `tab` can be removed from the DOM using any regular method (e.g., `removeChild`). In order for the removal animation to be shown, the `hideTab` helper method in `d2l-tabs` must be used. An example of this is shown below, which waits for the promise in `hideTab` to resolve and then removes the tab and panel from the DOM. Ensure that when a `tab` is removed, the corresponding panel is also removed.
+
+```
+const tabs = this.shadowRoot.querySelector('d2l-tabs');
+const tab = tabs.querySelector('d2l-tab');
+Promise.resolve(tabs.hideTab((tab))).then(() => {
+  const panel = tabs.querySelector(`d2l-tab-panel[labelled-by="${tab.id}"]`);
+  if (panel) tabs.removeChild(panel);
+  tabs.removeChild(tab);
+});
+```
+
+### Custom Tabs
+
+The `TabMixin` can be used to create custom tabs. It is IMPORTANT to call the `dispatchContentChangeEvent` function in `TabMixin` when content changes in the consumer in order to properly trigger calculations. Ensure that there is only one focusable element in any custom tab, else the focus and keyboard navigation behaviors becomes confusing for users.
```

Comment:
Since the `TabMixin` applies the `role` and `tabindex` to the host element, I wonder if we should explicitly state that consumers should not render focusable elements themselves. Thoughts?

Reply:
That's likely a good idea. Will update.